### PR TITLE
WIP: Compile helios in a canister

### PR DESCRIPTION
**Pull request by oblique**

### CODE REVIEW

File: client/Cargo.toml [hunk 1]
```diff
@@ -23,7 +23,6 @@ jsonrpsee = { version = "0.15.1", features = ["full"] }
 tokio = { version = "1", features = ["full"] }
 
 [target.'cfg(target_arch = "wasm32")'.dependencies]
-gloo-timers = "0.2.6"
-wasm-bindgen-futures = "0.4.33"
+ic-cdk-timers = "0.4.0"
+ic-cdk = "0.10.0"
 tokio = { version = "1", features = ["sync"] }
```
Comment (zvolin) [inline client/Cargo.toml:28]: those 2 don't need to be target guarded, and I don't get why is there a tokio?
anyway I'd just remove the whole client in a future so this is just a note

File: common/src/http.rs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+use crate::errors::HttpError;
+use ic_cdk::api::management_canister::http_request::HttpResponse;
+
+pub async fn http_get(url: &str) -> Result<HttpResponse, HttpError> {
+    let req = ic_http::create_request().get(url).build();
+    // TODO: should we pass cycles in http_get method or we should have a default one?
```
Comment (zvolin) [inline common/src/http.rs:6]: let's leave this todo as for now, we'll figure out later when things are working

File: consensus/Cargo.toml [hunk 3]
```diff
@@ -30,4 +29,3 @@ tokio = { version = "1", features = ["full"] }
 
 [target.'cfg(target_arch = "wasm32")'.dependencies]
 wasm-timer = "0.2.5"
```
Comment (zvolin) [inline consensus/Cargo.toml]: We'll need to replace usage of wasm-timer with a call to `ic_cdk::api::time()` or the `ic0` crate.
We can do that in next PR's, if so please add todo

**Review by zvolin — APPROVED** [review]
Nice first steps, I'm okay with merging it as a WIP

File: consensus/Cargo.toml (hunk 3 above)
Comment (zvolin) [reply consensus/Cargo.toml]: also just to note, wasm-timer is abandoned and author is against using it https://github.com/tomaka/wasm-timer/issues/21#issuecomment-1380544611

File: client/Cargo.toml (hunk 1 above)
Comment (oblique) [reply client/Cargo.toml:28]: tokio sync works in wasm32. client uses `Arc<RwLock<T>>` in some places. why `ic-cdk` don't need to be target guarded?

Comment (zvolin) [reply client/Cargo.toml:28]: it's written in a way that compiles on host, so you get an lsp support

**Review by zvolin — APPROVED** [review]

**Pull request merged**